src/ipc/file-system-get-entries.ts

```diff
@@ -0,0 +1,63 @@
+import makeDebug from 'debug'
+import { IpcMain, IpcMainInvokeEvent } from 'electron'
+import { promises as fs } from 'fs'
+import { isAbsolute, join } from 'path'
+import { assertHasReadAccess, HostFilePermission } from '../utils/access'
+import { Options } from '../utils/options'
+
+const debug = makeDebug('kiosk-browser:file-system-get-entries')
+
+export const channel = 'file-system-get-entries'
+
+export interface FileSystemEntry {
+  readonly name: string
+  readonly path: string
+  readonly isDirectory: boolean
+  readonly isFile: boolean
+  readonly size: number
+  readonly mtime: Date
+  readonly atime: Date
+  readonly ctime: Date
+}
+
+/**
+ * Get entries for a directory, includes stat information for each entry.
+ */
+export async function getEntries(
+  permissions: readonly HostFilePermission[],
+  hostname: string,
+  path: string,
+): Promise<FileSystemEntry[] | undefined> {
```

Comment:
I like this function signature that requires permissions and context.

Reply:
and it makes me realize that we probably should have permissions based on web origin, not hostname. So `http://locahost:3000` vs. `localhost`. I realize this requires changing the underlying code that is permissionsed by hostname instead of origin.

Reply:
Done.